app/models/work_packages/scopes/relatable.rb

```diff
@@ -40,33 +40,114 @@ module Relatable
-      #   aunt and nephew.
+      #   aunt and nephew. Additionally, a transitive relationship which leaves the tree for any one hop, is exempt from that rule
+      #   so it is possible to have child --- blocks ---> other --- blocks ---> parent.
       # * No circles between trees: The ancestor/descendant chain is considered bidirectional when calculating relatability.
       #   This means that starting from the work package both the descendants as well as the ancestors are considered. That
       #   way, relations like
-      #     Child 1 --- parent ---> Parent 1 ---> follows ---> Parent 2 ---> child ---> Child 2 --- follows ---> Child 1
+      #     Child tree 1
+      #       --- parent ---> Parent tree 1
+      #       --- follows ---> Parent tree 2
+      #       --- child ---> Child tree 2
+      #       --- follows ---> Child tree 1
       #   are prevented.
       #
+      # For some of the relations, this has actual relevance (FOLLOWS, PRECEDES, PARENT because of scheduling) while for the
+      # most, it is simply a question of semantics which is not manifested in code (all the other).
+      #
       # For the sake of this scope, the parent relation (Relation::TYPE_PARENT) is also included in the list of relation_types
       # even though it is not stored in the same data structure. All Relation::TYPE_* values can be provided even those that
       # are not canonical, e.g. Relation::TYPE_PRECEDES. The calculation for those relation types are then inverted and the
       # canonical type is used, e.g. Relation::TYPE_FOLLOWS.
       #
       # There are a couple of exceptions and additions to the limitations outlined above for the following types:
-      # * Relation::TYPE_RELATES: Since this is essentially unrelated and does not carry a lot of semantic, the work packages
+      # * Relation::TYPE_RELATES: Since this is essentially undirected and does not carry a lot of semantic, the work packages
       #   are simply somehow related, such relations do not follow the "non circular" nor the "ancestor/descendant" rule.
       # * Relation::TYPE_PARENT: Since creating a new relationship will remove the old parent relationship, current ancestors
       #   (except the direct parent) are relatable to. Descendants however are not since that would create a circle.
+      #   In addition to the existing hierarchy, the FOLLOWS relationships are taken into account. Predecessors and successors
+      #   (FOLLOWS relationship) may also not be related to via a PARENT relation. However, parents and children of those
+      #   predecessors/successors can be related to. But they still need to be considered in the code since they might be
+      #   part of an almost completed circle which would be closed if a work package is added as a parent. E.g. in
+      #     WP1 --- follows ---> WP2 --- parent ---> WP3 --- follows ---> WP4
+      #   WP4 is not a valid parent candidate whereas WP3 is.
```

Comment:
A valid parent candidate for which work package? WP1?
But if WP3 is a child of WP2, I do not understand why WP1 can be a child of WP3. That should not be possible. What am I getting wrong here?

Reply:
Yes, the queried for work package would be WP1. I should have made that clearer in the text.

There is a flaw in my example here in that the parent relation between WP2 and WP3 means to indicate that WP3 is a parent of WP2 and not the other way around. That is why WP1 can become a child of WP3 which would turn its predecessor WP2 into a sibling. But WP4 cannot be a parent since that would result in 

```
WP3 --> WP4
 |       |
 v       v
WP2 <-- WP1

```

I suggest to fix the flaw and also to show the resulting trees which might make it more understandable.

Reply:
Done in a29cf73bb2ea3ba08594b53051c37312639b1465